controllers/hyperparametertuningjob/hpo_trainingjob_spawner.go

```diff
@@ -80,14 +80,19 @@ func (s hpoTrainingJobSpawner) SpawnMissingTrainingJobs(ctx context.Context, hpo
 	k8sNamespace := hpoJob.ObjectMeta.GetNamespace()
 	awsRegion := *hpoJob.Spec.Region
 	sageMakerEndpoint := hpoJob.Spec.SageMakerEndpoint
-
-	paginator := s.SageMakerClient.ListTrainingJobsForHyperParameterTuningJob(ctx, hpoJobName)
+	var nextToken *string = nil
 
 	// WaitGroup allowing us to do checks in parallel.
 	var wg sync.WaitGroup
 
-	for paginator.Next(ctx) {
-		list := paginator.CurrentPage()
+	for {
```

Comment:
Where is the condition for this `for`? Is this the equivalent of a `while True`?

Reply:
yep